hr_timesheet_sheet_worked_time/models/hr_timesheet_sheet_sheet.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import datetime
+from odoo import models, fields, api
+
+
+class HrTimesheetSheet(models.Model):
+    _inherit = 'hr_timesheet_sheet.sheet'
+
+    standard_work_hours = fields.Float(
+        compute='_compute_standard_work_hours',
+        string="Standard Work Hours",
+    )
+    expected_work_hours = fields.Float(
+        compute='_compute_expected_work_hours',
+        string="Expected Work Hours",
+    )
+    overtime_hours = fields.Float(
+        compute='_compute_overtime_hours',
+        string="Overtime Hours",
+        store=False
+    )
+
+    @api.depends('standard_work_hours')
+    def _compute_expected_work_hours(self):
+        for sheet in self:
+            public_holidays = self.env['hr.holidays.public'].search(
+                [('year', '=', datetime.now().year),
+                 ('country_id', '=', self.user_id.partner_id.country_id.id)]
+            )
+            holiday_hours = 0.0
+            for line in public_holidays.mapped('line_ids'):
+                public_holiday = \
+                    sheet.employee_id.calendar_id.attendance_ids.filtered(
+                        lambda attendance: attendance.date_from == line.date or
+                                           attendance.date_to == line.date)
+                if public_holiday:
+                    for attendance in public_holiday:
+                        holiday_hours += \
+                            attendance.hour_from - attendance.hour_to
+            sheet.expected_work_hours = \
+                self.standard_work_hours - abs(holiday_hours)
```

Comment:
Let's have a separate field to compute the holiday hours.
And the holiday hours should be calculate by searching `hr.holidays.public` between timesheet's `date_from` to `datetime.now()`.
Then caluclate the holiday hours depends on the `dayofweek` in the `attendance_ids`.
As the `date_to` and `date_from` in the `calendar_id.attendance_ids` will be empty